### PR TITLE
ci: add go-pr-testing workflow

### PR DESCRIPTION
**Pull request by smoya**
Part of https://github.com/asyncapi/shape-up-process/issues/94
This PR adds a new workflow for **Go** projects. It runs two steps:

- **lint**
  - It runs [golangci-lint](https://golangci-lint.run/) linter, which it is a tool that runs the most common linters for go.
  - Note: The configuration for the linters are up to each repository (via `.golangci.yml` file), however I think it makes sense if we create a repository template for Go projects that contain, among others, that file. I created a [gist](https://gist.github.com/smoya/35720dd325025dc5665a4f86d1849aae) containing all the config we might want to set.
- **test**
  - It ensures the app can be built.
  - It runs tests.

Note: The golangci-lint gh action not only fails the build in case of lint issues but it shows those issues in the PR review as a check runs. 
Example:
image

### CODE REVIEW

**Review by jonaslagoni — DISMISSED** [review]
Looks fined to me, but we might need to start consider another approach in regards to sharing global workflow for multiple languages 🤔 In the end it could be a lot of checks 😅

File: .github/workflows/if-go-pr-testing.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
+#This action is centrally managed in https://github.com/asyncapi/.github/
+#Don't make changes to this file in this repo as they will be overwritten with changes made to the same file in above mentioned repo
+#It does magic only if there is package.json file in the root of the project
```
Comment (derberg) [inline .github/workflows/if-go-pr-testing.yml:3]: go.mod please and not package.json

**Review by derberg — CHANGES_REQUESTED** [review]
Once you make a small fix I approve as all is good.

File: .github/workflows/if-go-pr-testing.yml (hunk 1 above)
Comment (smoya) [reply .github/workflows/if-go-pr-testing.yml]: Fixed!

**Review by derberg — APPROVED** [review]

**Pull request merged**